File: 프로그래머스/0/181866. 문자열 잘라서 정렬하기/문자열 잘라서 정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> solution(string myString) {
    vector<string> answer;
    string tmp="";
    
    for(char c:myString)
    {
        if(c=='x')
        {
            if(!tmp.empty())
            {
                answer.push_back(tmp);
                tmp="";
            }
        }
        else
            tmp+=c;
    }
    if(!tmp.empty())
        answer.push_back(tmp);
    
    sort(answer.begin(),answer.end());
    
    return answer;
}

File: 프로그래머스/0/181866. 문자열 잘라서 정렬하기/문자열 잘라서 정렬하기.py
def solution(myString):
    answer, tmp = [], []
    
    for char in myString:
        if char != 'x':
            tmp.append(char)
        else:
            if tmp:  # Check if tmp is not empty
                answer.append(''.join(tmp))  # Join the characters in tmp and append to answer
                tmp = []  # Reset tmp
    
    # If there's any remaining characters in tmp after the loop
    if tmp:
        answer.append(''.join(tmp))
    
    # Sort the list
    answer.sort()
    
    return answer
